distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso

```diff
@@ -0,0 +1,582 @@
+        handle_java_exception <|
+            case math_context of
+                Nothing -> Decimal.Value (self.big_decimal.divide that.big_decimal)
+                _ -> Decimal.Value (self.big_decimal.divide that.big_decimal math_context.math_context)
+
+    ## GROUP Operators
+       ICON math
+       Divides a `Decimal` by another `Decimal` or other kind of number, or
+       divides another kind of number by a `Decimal`.
+
+       Arguments:
+       - that: The number to divide by this.
+
+       Division in Enso will undergo automatic conversions such that you need
+       not convert other numeric types to `Decimal` manually.
+
+       > Example
+         Dividing 1065.9378 by 23.34.
+
+             a = Decimal.new "1065.9378"
+             b = Decimal.new "23.34"
+             c = a / b
+             # => Decimal.new 45.67
+    / : Decimal -> Decimal -> Decimal
+    / self (that:Decimal) = self.divide that
+
+    ## GROUP Operators
+       Compute the negation of this.
+
+       > Example
+         Negate 5.1 to get -5.1.
+
+             5.1.negate
+             # => Decimal.new -5.1
+    negate : Decimal
+    negate self = Decimal.Value self.big_decimal.negate
+
+    # PRIVATE
+    get_precision : Integer
+    get_precision self = self.big_decimal.precision
+
+    # PRIVATE
+    get_scale : Integer
+    get_scale self = self.big_decimal.scale
+
+    # PRIVATE
+    get_unscaled_value : Integer
+    # TODO: This +0 is needed to allow internal represeantations to be compared.
+    get_unscaled_value self = self.big_decimal.unscaledValue + 0
+
+    # PRIVATE
+    get_internal_representation : [Integer]
+    get_internal_representation self = [self.get_unscaled_value, self.get_precision, self.get_scale]
+
+    ## PRIVATE
+    to_text : Text
+    to_text self = self.big_decimal.toString
+
+    ## PRIVATE
+    to_text_without_scientific_notation : Text
+    to_text_without_scientific_notation  self = self.big_decimal.toPlainString
+
+## PRIVATE
+handle_number_format_exception ~action =
+    Panic.catch NumberFormatException action caught_panic->
+        Error.throw (Number_Parse_Error.Error caught_panic.payload.getMessage)
+
+## PRIVATE
+handle_precision_loss : Any -> Conversion_Result -> Any ! Loss_Of_Numeric_Precision
+handle_precision_loss original_value conversion_result:Conversion_Result -> Any ! Loss_Of_Numeric_Precision =
+    if conversion_result.hasPrecisionLoss.not then conversion_result.newValue else
+        new_value = conversion_result.newValue
+        Warning.attach (Loss_Of_Numeric_Precision.Warning original_value new_value) new_value
+
+attach_loss_of_numeric_precision : Float -> Any -> Any
+attach_loss_of_numeric_precision f value =
+    Warning.attach (Loss_Of_Numeric_Precision.Warning f f) value
+
+handle_java_exception ~action =
+    Panic.catch ArithmeticException action caught_panic->
+        Error.throw (Arithmetic_Error.Error caught_panic.payload.getMessage)
+
+type Decimal_Comparator
```

Comment:
Please tag this and comparable conversions as private.

Reply:
Done.

Reply:
Cannot the comparator be moved into own `.enso` file that is marked as `private` to use capabilities provided by #7840?